src/memoization/memoization.jl

```diff
@@ -0,0 +1,46 @@
+mutable struct MemoNLP <: AbstractNLPModel
+  model :: AbstractNLPModel
+  meta :: NLPModelMeta
+  counters :: Counters
+  min_elm :: Int
+  max_elm :: Int
+  memo :: Dict{AbstractVector, Any}
```

Comment:
"People" say it's not a good idea to have abstract types inside a struct. Also, wouldn't the value type be an `AbstractFloat`? Or do you have something more general in mind?

Reply:
Yeah it definitely should just be ``` AbstractFloat```, thank you.